src/python/pants/backend/python/goals/coverage_py.py

```diff
@@ -314,9 +314,7 @@ async def generate_coverage_reports(
         pex_processes.append(
             PexProcess(
                 coverage_setup.pex,
-                # We pass `--ignore-errors` because Pants dynamically injects missing `__init__.py`
-                # files and this will cause Coverage to fail.
-                argv=(report_type.report_name, "--ignore-errors"),
+                argv=(report_type.report_name,),
```

Comment:
Removing `--ignore-errors` without the above fix causes the integration tests to fail, so I think that the integration tests are already a good enough regression test for this edge case.